src/modules/sync/sync/collections.ts

```diff
@@ -0,0 +1,33 @@
+import { CollectionModels, ManagementClient } from "@kontent-ai/management-sdk";
+
+import { DiffModel } from "../types/diffModel.js";
+import { getTargetCodename, PatchOperation } from "../types/patchOperation.js";
+
+export const syncCollections = (client: ManagementClient, collections: DiffModel["collections"]) => {
+  if (!collections.length) {
+    return Promise.resolve();
+  }
+
+  return client
+    .setCollections()
+    .withData(collections.map(transformCollectionsReferences))
+    .toPromise();
+};
+
+const transformCollectionsReferences = (operation: PatchOperation): CollectionModels.ISetCollectionData => {
+  const pathParts = operation.path.split("/");
+  const propertyName = pathParts[pathParts.length - 1];
+  const codename = getTargetCodename(operation);
+
+  return {
+    ...operation,
```

Comment:
Use the `omit` function instead of setting the unwanted properties to `undefined`.